crates/mun_codegen/src/ir/types.rs

```diff
@@ -7,6 +8,10 @@ impl<'ink> TransparentValue<'ink> for abi::Guid {
     fn as_target_value(&self, context: &IrValueContext<'ink, '_, '_>) -> Value<'ink, Self::Target> {
         self.0.as_value(context)
     }
+
+    fn as_bytes_and_ptrs(&self, _: &IrValueContext<'ink, '_, '_>) -> Vec<BytesOrPtr<'ink>> {
+        vec![BytesOrPtr::Bytes(self.0.to_vec())]
```

Comment:
Would it make sense to implement a `From<Vec<u8>>` for `BytesOrPtr`?

Reply:
That is actually implemented. I'll change this instance.